website/src/_posts/2021-08-30.31.md

```diff
@@ -0,0 +1,250 @@
+
+- [abortcontroller-polyfill](https://github.com/mo/abortcontroller-polyfill)
+- [core-js](https://github.com/zloirock/core-js)
+- [md-gum-polyfill](https://github.com/mozdevs/mediaDevices-getUserMedia-polyfill)
+- [resize-observer-polyfill](https://github.com/que-etc/resize-observer-polyfill)
+- [whatwg-fetch](https://github.com/github/fetch)
+
+If you're using a bundler, you need import them before Uppy:
+
+```js
+import 'core-js'
+import 'whatwg-fetch'
+import 'abortcontroller-polyfill/dist/polyfill-patch-fetch'
+// Order matters: AbortController needs fetch which needs Promise (provided by core-js).
+
+import 'md-gum-polyfill'
+import ResizeObserver from 'resize-observer-polyfill'
+
+window.ResizeObserver ??= ResizeObserver
+
+export { default } from '@uppy/core'
+export * from '@uppy/core'
+```
+
+If you're using Uppy from a CDN, we now provide two bundles, one for up-to-date browsers that do not include polyfills and use modern syntax, and one for legacy browsers. When migrating, be mindful about which browsers you want to support:
+
+```html
+<!-- Modern browsers (recommended) -->
+<script src="https://releases.transloadit.com/uppy/v2.0.1/uppy.min.js"></script>
+
+<!-- Legacy browsers (IE11+) -->
+<script nomodule src="https://releases.transloadit.com/uppy/v2.0.1/uppy.legacy.min.js"></script>
+<script type="module">import "https://releases.transloadit.com/uppy/v2.0.1/uppy.min.js";</script>
+```
+
+Please note that while you may be able to get 2.0 to work in IE11 this way, we do not officially support it anymore.
+
+## Faster
+
+Uppy now loads faster due to the decreased bundle size, and compared to 1.0, adding many files (hundreds or thousands), used to take dozens of seconds, which has now been reduced to milliseconds. So for that use case, speed improvements of 1000X are not uncommon or exaggerated.
+
+This was made possible by avoiding rerendering all the file components whenever something changes, using `memoize` and `virtual-list` (allows us to only render what is actually on-screen). In addition, multiple files are now added to state in one go via `uppy.addFiles(Array)` vs when `uppy.addFile(File)` + `uppy.setState` were being called in a loop.
+
+Before optimizations:
+
+![](/images/blog/2.0/uppy-performance-before.gif)
+
+After:
+
+![](/images/blog/2.0/uppy-performance-after.gif)
+
+## Preact X and upgraded dependencies
+
+We’ve upgraded almost all of Uppy’s dependencies. This includes the migration to Preact X. All plugins that depend on Preact have been upgraded from `8.2.9` to the latest version `10.5.13`. If you’d like your custom plugin to continue working with Uppy 2.0, it also needs to be using latest Preact now.
+
+Other notable upgrades include browserify to `v10`, typescript to `v4.3`, autoprefixer to `v10`, and lerna to `v4`.
```

Comment:
Not sure users would or should care about lerna, AFAIK it's only relevant for Uppy development and doesn't affect users.
```suggestion
Other notable upgrades include browserify to `v10`, typescript to `v4.3`, and autoprefixer to `v10`.
```

Reply:
@arturi could you apply this one also please?